From 04cd4e8792123ff1953bd4cedf2dde18ed4a8781 Mon Sep 17 00:00:00 2001
From: Arik Nemtsov <arik@wizery.com>
Date: Wed, 18 May 2011 06:32:52 +0300
Subject: [PATCH 61/75] wl12xx: configure BA initiator policy with current role_id

It appears the ANY role_id is not very well supported.
---
 drivers/net/wireless/wl12xx/acx.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/wl12xx/acx.c b/drivers/net/wireless/wl12xx/acx.c
index c148bce..2cf34db 100644
--- a/drivers/net/wireless/wl12xx/acx.c
+++ b/drivers/net/wireless/wl12xx/acx.c
@@ -1415,8 +1415,8 @@ int wl1271_acx_set_ba_initiator_policy(struct wl1271 *wl)
 		goto out;
 	}
 
-	/* enable for ANY role */
-	acx->role_id = 0xFF;
+	/* set for the current role */
+	acx->role_id = wl->role_id;
 	acx->tid_bitmap = wl->conf.ht.tx_ba_tid_bitmap;
 	acx->win_size = wl->conf.ht.tx_ba_win_size;
 	acx->inactivity_timeout = wl->conf.ht.inactivity_timeout;
-- 
1.7.0.4

